amethyst_assets/src/prefab/system.rs

```diff
@@ -102,26 +106,32 @@ where
                     let new_entity = entities.create();
                     self.entities.push(new_entity);
                     if let Some(parent) = entity_data.parent {
-                        parents
-                            .insert(
-                                new_entity,
-                                Parent {
-                                    entity: self.entities[parent],
-                                },
-                            ).unwrap();
+                        if let Err(e) = parents.insert(
+                            new_entity,
+                            Parent {
+                                entity: self.entities[parent],
+                            },
+                        ) {
+                            error!("Unable to insert new entity with a `Parent`: {}", e);
+                        }
+                    }
+                    if let Err(e) = tags.insert(
+                        new_entity,
+                        PrefabTag::new(prefab.tag.expect("Prefab tag is not present")),
+                    ) {
+                        error!("Unable to insert new entity with a `PrefabTag`: {}", e);
                     }
-                    tags.insert(new_entity, PrefabTag::new(prefab.tag.unwrap()))
-                        .unwrap();
                 }
                 // create components
                 for (index, entity_data) in prefab.entities.iter().enumerate() {
                     if let Some(ref prefab_data) = &entity_data.data {
-                        prefab_data
-                            .add_to_entity(
-                                self.entities[index],
-                                &mut prefab_system_data,
-                                &self.entities,
-                            ).unwrap();
+                        if let Err(e) = prefab_data.add_to_entity(
+                            self.entities[index],
+                            &mut prefab_system_data,
+                            &self.entities,
+                        ) {
+                            error!("Unable to insert prefab system data for entity: {}", e);
```

Comment:
Again, could this happen and not be a bug?
If so, it might be beneficial to have the engine crash with a meaningful message.

Reply:
This one aswell. I haven't been able to figure out if its a bug or a possible occurence. Could I have some guidance perhaps?

Reply:
@Rhuagh I saw Moxinilian said that it was you who created the asset system. Could you help me understand if a `None` in this case is possible or would be a bug?